RadixWallet/Features/SettingsFeature/DisplayMnemonics/Coordinator/DisplayMnemonics.swift

```diff
@@ -190,3 +193,33 @@ extension DisplayEntitiesControlledByMnemonic.State {
 		self.isMnemonicMarkedAsBackedUp = true
 	}
 }
+
+// MARK: - DisplayMnemonics.ComparableEntities
+private extension DisplayMnemonics {
+	/// A helper struct to sor mnemonics using the following criteria:
+	/// 1) First should be the one associated with main device factor source.
+	/// 2) Last should be those with Olympia device factor source.
+	/// 3) In the middle will show those with babylon device factor source sorted by date added.
+	struct ComparableEntities: Comparable {
+		let state: DisplayEntitiesControlledByMnemonic.State
+		let deviceFactorSource: DeviceFactorSource
+
+		static func < (lhs: Self, rhs: Self) -> Bool {
+			let lhs = lhs.deviceFactorSource
+			let rhs = rhs.deviceFactorSource
+			if lhs.isExplicitMain {
+				return true
+			} else if rhs.isExplicitMain {
+				return false
+			} else {
+				if lhs.isBDFS, rhs.isBDFS {
+					return lhs.common.addedOn < rhs.common.addedOn
+				} else if lhs.isBDFS {
```

Comment:
The is last if else can be:

`else { return lhs.isBDFS }` right?

Reply:
🤦 
thanks!